geps/gep-713/index.md

```diff
@@ -3,1155 +3,1172 @@
+* Patch Overrides
 
-Which again, come down to discoverability, and can probably be addressed in similar
-ways at an API level to the Application Developer's concerns.
+Additionally, implementations MAY specify _custom_ merge strategies. These are implementation-specific strategies where the specs of 2 metaresources are mixed into a composition of both, following a custom merge algorithm specified by the metaresource kind.
 
-An important note here is that a key piece of information for Policy Admins and
-Cluster Operators is “How many things does this Policy affect?”. In the parable,
-this would have enabled Chihiro to know that deleting the Namespace Policy would
-affect many other people than just Ana.
+##### Rules for implementing merge strategies
 
-### Problems we need to solve
+Metaresource kinds MAY opt to implement any of the well-known or custom merge strategies, including multiple strategies by the same metaresource kind.
 
-Before we can get into solutions, we need to discuss the problems that solutions
-may need to solve, so that we have some criteria for evaluating those solutions.
+Metaresource kinds that do not specify any merge strategy and only support targeting a single kind (with Declared target = Effective target), by default MUST implement the **None** merge strategy. (See the definition of [Direct](#direct) class of metaresources.)
 
-#### User discoverability
+Metaresource kinds that do not specify any merge strategy and support targeting multiple effective kinds, by default MUST implement the **Atomic Defaults** merge strategy.
 
-Let's go through the various users of Gateway API and what they need to know about
-Policy Attachment.
+Metaresource kinds that implement more than one merge strategy MUST define a clear structure for the metaresource CRs to specify which one of the supported strategies to apply. Metaresource CRs MUST NOT be allowed to declare more than one merge strategy at a time, but only one of the supported strategies. If no merge strategy is specified by a given metaresource CR, the **Atomic Defaults** merge strategy SHOULD be assumed, provided it's one of the supported merge strategies implemented by the metaresource kind.
 
-In all of these cases, we should aim to keep the troubleshooting distance low;
-that is, that there should be a minimum of hops required between objects from the
-one owned by the user to the one responsible for a setting.
+For metaresource kinds that support merge strategy specified by the user at the CR, the following rules, continuing on ties, MUST be implemented to determine which merge strategy to apply between two metaresources in conflict:
+1. Between two metaresources targeting at different levels of the hierarchy, the one attached higher (less specific) dictates the merge strategy to use to resolve the conflict.
+2. Between two metaresources targeting at the same level of the hierarchy, the older metaresource based on creation timestamp dictates the merge strategy.
```

Comment:
Defaults/Overrides doesn't make sense for policies applied to the same level. This should either be None (no merging, oldest wins), or a compositional merge that merges newer policies into older ones in order of priority from old(high)->new(low) without invalidating older policies.